### PR TITLE
Some docs

### PR DESCRIPTION
**Pull request by elisehuard**
docs that were originally in mesos-setup project, or that needed writing down.  This should grow while we work on it (probably especially the troubleshooting section)

### CODE REVIEW

File: doc/configuration.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+# Configuration
+
+## What does terraboot do
+
+It generates [terraform]()-readable json. Make sure you have latest terraform installed, they do tend to fix bugs from one version to the other
```
Comment (acron0) [inline doc/configuration.md:5]: wants a link :)

File: doc/configuration.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+## Modules
+
+terraboot is divided in 4 modules at the moment:
+
+* vpc (generated by the _vpc-vpn-infra_ function): for vpc, subnets, NAT, vpn, ELK, monitoring and alerting boxes (the setup of the boxes is not automated at this point, see install-dns, install-icinga, install-influx, install-logstash)
```
Comment (acron0) [inline doc/configuration.md:11]: Use backtick to isolate fn names.

**Comment by jasebell** [comment]
lgtm :shipit:

File: doc/configuration.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+* vpc dns (generated by the _vpc-public-dns_ function): for a number of public DNS names on the VPC (requires having a public domain on Route53).
+* cluster (geenrated by _cluster-infra_ function): for individual clusters - the idea being that you can have several clusters per vpn.
+* cluster dns (generated by _cluster-publlic-dns_ function): for cluster-specific DNS (also requires having a public domain on Route53)
+
+There are two main configuration files: a edn files per module for fixed details, and more variable parameters (like instance types or open ports) in the calls to terraboot.
```
Comment (acron0) [inline doc/configuration.md:16]: Find this sentence a little confusing.

File: doc/configuration.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+## edn files
+
+This should contain details which are mostly fixed over the whole infrastructure. One of these file is passed in as an argument to `lein run`, for instance `lein run resources/terraboot-staging.edn`.
+
+    {:region "your-aws-region"
+     :bucket-name "your-s3-bucket"
+     :aws-profile "your-aws-profile"
+     :account-number "your-aws-account-number" ;; to generate ARN
+     :azs [:a :b :c]
```
Comment (acron0) [inline doc/configuration.md:26]: Add examples for `:azs` and `:target`

File: doc/troubleshooting.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+### Is the slave disk full (without mesos knowning) because of docker?
+
+Log on into the slave (ssh core@<ip> when on the VPN)
+Check for disk space `df -h` - if the main disk is at 100%, this may be your problem.
+Solution
+
+    docker rmi $(docker images -a -q)
+    docker rm $(docker ps -a -q)`
+
+If it doesn't release space properly, stop docker, rm -rf /var/lib/docker and restart (but this means pulling all images again).
```
Comment (acron0) [inline doc/troubleshooting.md:25]: Use backticks around bash command

Comment (elisehuard) [reply doc/troubleshooting.md:25]: indented by 4 spaces has same effect for markdown (https://github.com/MastodonC/terraboot/blob/master/doc/troubleshooting.md)

**Pull request merged**